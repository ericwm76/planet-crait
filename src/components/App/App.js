import React, { Component } from 'react';
import './App.scss';
import { Route, Nav} from 'react-router-dom';
import { getMovieData, getCharacters } from '../../apiCalls';
import sampleCharacters from '../../SampleCharacterData'

import MoviesContainer from '../MoviesContainer/MoviesContainer';
import CharactersContainer from '../CharactersContainer/CharactersContainer';
import FavouritesContainer from '../FavouritesContainer/FavouritesContainer';
import LandingPage from '../LandingPage/LandingPage';
import ScrollText from '../ScrollText/ScrollText';


class App extends Component { 
  constructor() {
    super();
    this.state = {
       movie: {},
       characters: [],
       name: '',
       quote: '',
       rank: '',
       crawl: ''
    }
  }

  selectMovie = (movie) => {
    getMovieData(movie)
     .then(data => {
      this.setState({ movie : data, crawl : data.opening_crawl})
      
      getCharacters(this.state.movie.characters)
        .then(data => this.setState({ characters: data }))
        .then(console.log(this.state.characters))
    })
  }

  getSampleCharacters = () => {
    this.setState(() => ({ characters: sampleCharacters }));
  }

  setUsers = (name, quote, rank) => {
    this.setState({
      name: name,
      quote: quote,
      rank: rank
    })
  }

  render() {
    return (
      <main className="App">
        <Route exact path='/' render={() => <LandingPage setUsers={this.setUsers} />} />
        <Route exact path='/movies' render={() => <MoviesContainer selectMovie={this.selectMovie} name={this.state.name} quote={this.state.quote} rank={this.state.rank} />} />
        <Route exact path='/movies/:id' render={() => <CharactersContainer characters={this.state.characters} name={this.state.name} quote={this.state.quote} rank={this.state.rank}/>} />
        <Route exact path='/favourites' render={() => <FavouritesContainer name={this.state.name} quote={this.state.quote} rank={this.state.rank}/>} />
        <ScrollText movie={this.state.movie}/>
      </main>
    );
  }
}

export default App;