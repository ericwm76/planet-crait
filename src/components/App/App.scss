.App {
  text-align: center;
  background-image: url('https://viajento.files.wordpress.com/2016/10/salar-de-uyuni-bolicc81via-deserto.jpg?w=640');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: top;
}

