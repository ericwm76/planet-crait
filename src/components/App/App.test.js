import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App',() => {
  let wrapper;
  let mockMovieData;
  let mockCharacters;

  beforeEach(() => {
    wrapper = shallow(<App />);
    mockMovieData = {
      url: "https://swapi.co/api/films/4/",
      episode_id: 1,
      title: "The Phantom Menace",
      release_date: "1999",
      img: "https://images-na.ssl-images-amazon.com/images/I/51o6zchqOFL.jpg"
    };

    mockCharacters = [
      {name: "Luke Skywalker", species: "Human", homeworldName: "Tatooine", homeworldPop: "200000", appearsIn: ["The Empire Strikes Back", "Return of the Jedi", "A New Hope", "Revenge of the Sith", "The Force Awakens"], favorited: false},
      {name: "C-3PO", species: "Droid", homeworldName: "Tatooine", homeworldPop: "200000", appearsIn: ["The Empire Strikes Back", "Return of the Jedi", "A New Hope", "The Phantom Menace", "Attack of the Clones", "Revenge of the Sith"], favorited: false}
      ];
  })

  it('should match snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should update the state of name, quote, and rank', () => {

    wrapper.instance().setUsers('Conchelo', 'Do.Or Do Not', 'expert');

    expect(wrapper.state('name')).toEqual('Conchelo');
    expect(wrapper.state('quote')).toEqual('Do.Or Do Not');
    expect(wrapper.state('rank')).toEqual('expert');
  });

   it('should update state when selectMovie is invoked', () => {
     // mock getMovieData function
     // mock getCharacters function
     // invoke the function
     // expect wrapper.state to equal to the mocks
     
    //  wrapper.instance().getMovieData = jest.fn();
    //  wrapper.instance().getCharacters = jest.fn();

      wrapper.instance().selectMovie(mockMovieData);

      
   });

   it('should call getMovieData when selectMovie is called', () => {
     wrapper.instance().selectMovie(mockMovieData);

     let mockMovieMethod = jest.fn();

     expect(wrapper.getMovieData).toHaveBeenCalled();
   })
}) 
