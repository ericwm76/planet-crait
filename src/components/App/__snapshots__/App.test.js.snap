// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot with all data passed in correctly 1`] = `
<main
  className="App"
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/:id"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/favourites"
    render={[Function]}
  />
</main>
`;
